ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleBannerAdapter.java

```diff
@@ -0,0 +1,256 @@
+            mVungleManager.removeActiveBannerAd(mPlacementId);
+            if (mPendingRequestBanner && mVungleListener != null) {
+                mVungleListener.onAdFailedToLoad(AdRequest.ERROR_CODE_NO_FILL);
+            }
+        }
+    };
+
+    private PlayAdCallback mAdPlayCallback = new PlayAdCallback() {
+        @Override
+        public void onAdStart(String placementId) {
+            if (mPendingRequestBanner && mVungleListener != null) {
+                mVungleListener.onAdStart(placementId);
+            }
+        }
+
+        @Override
+        public void onAdEnd(String placementId, boolean completed, boolean isCTAClicked) {
+            if (mPendingRequestBanner && mVungleListener != null) {
+                mVungleListener.onAdEnd(placementId, completed, isCTAClicked);
+            }
+        }
+
+        @Override
+        public void onError(String placementId, VungleException exception) {
+            Log.d(TAG, "Ad play failed:" + VungleBannerAdapter.this);
+            mIsLoading.set(false);
+            mVungleManager.removeActiveBannerAd(mPlacementId);
+            if (mPendingRequestBanner && mVungleListener != null) {
+                mVungleListener.onAdFail(placementId);
+            }
+        }
+    };
+
+    private void loadBanner() {
+        if (mVungleManager.isAdPlayable(mPlacementId, mAdConfig.getAdSize())) {
+            createBanner();
+        } else if (mVungleManager.isValidPlacement(mPlacementId)) {
+            if (AdConfig.AdSize.isBannerAdSize(mAdConfig.getAdSize())) {
+                Banners.loadBanner(mPlacementId, mAdConfig.getAdSize(), mAdLoadCallback);
+            } else {
+                Vungle.loadAd(mPlacementId, mAdLoadCallback);
+            }
+        } else { // passed Placement Id is not what Vungle's SDK gets back after init/config
+            mIsLoading.set(false);
+            mVungleManager.removeActiveBannerAd(mPlacementId);
+            if (mVungleListener != null) {
+                mVungleListener.onAdFailedToLoad(AdRequest.ERROR_CODE_INVALID_REQUEST);
+            }
+        }
+    }
+
+    private void createBanner() {
+        if (!mPendingRequestBanner) {
+            return;
+        }
+        cleanUp();
+        mIsLoading.set(false);
+        RelativeLayout.LayoutParams adParams = new RelativeLayout.LayoutParams(RelativeLayout.LayoutParams.WRAP_CONTENT,
+                RelativeLayout.LayoutParams.WRAP_CONTENT);
+        adParams.addRule(RelativeLayout.CENTER_HORIZONTAL, RelativeLayout.TRUE);
+        adParams.addRule(RelativeLayout.CENTER_VERTICAL, RelativeLayout.TRUE);
+        if (AdConfig.AdSize.isBannerAdSize(mAdConfig.getAdSize())) {
+            mVungleBannerAd = Banners.getBanner(mPlacementId, mAdConfig.getAdSize(), mAdPlayCallback);
+            if (mVungleBannerAd != null) {
+                updateVisibility(mVisibility);
+                mVungleBannerAd.setLayoutParams(adParams);
+                mAdLayout.addView(mVungleBannerAd);
```

Comment:
if the same AdRequest(equal requestUniqueId & VungleBannerAdapter instance) can be used I believe we have to remove mVungleBannerAd & mVungleNativeAd from their old mAdLayout here or in the cleanUp()

Reply:
yes, fixed.